### ISSUE TITLE
RemoveArrayAction does not update child controls ID

### ISSUE THREAD

**Issue by lucax88x** [MrWolfZ/ngrx-forms#41]
Ciao Wolfz :)

I'm handling an array of entities, and I have a remove button on each entity row, so for example, when I have 3 rows I can remove the middle one, and so on.

What's going on is that when I dispatch a RemoveArray action it's indeed updating the ID of the subsequents items, but not the id of their inner controls.

simple example is below:

1)
3 items on array, with nested fields of Name & Surname

```
    ITEM0
     Name0
     Surname0
    ITEM1
     Name1
     Surname1
    ITEM2
     Name2
     Surname2
```

2)

we now remove ITEM1 with a Remove_Array action from the library, ITEM2 gets correctly renamed to ITEM1, but the nested fields remains the same.

```
    ITEM0
     Name0
     Surname0
    ITEM1
     Name2
     Surname2
```

3)
we now add a 3rd item, it's get ITEM2, and the nested fields? guess it..

```
    ITEM0
     Name0
     Surname0
    ITEM1
     Name2
     Surname2
    ITEM2
     Name2
     Surname2
```

This situation now leads to duplicated data, because when I modify Name on ITEM1 also modifies the one in ITEM2.

Regards,
Luca

**Comment by MrWolfZ** [comment]
Thanks for finding and reporting this bug. I'll work on a fix asap.

**Comment by MrWolfZ** [comment]
I have just released version 2.3.0 which fixes this issue.

**Comment by lucax88x** [comment]
Works, thanks man 👍
